File: 3952-trionic-array-i/trionic-array-i.cs
public class Solution {
    public bool IsTrionic(int[] nums) {
        int ion = 1;
        int l = nums.Length;
        if ( l <= 3 || nums[0] > nums[1] || nums[l-2] > nums[l-1])
        return false;
        for(int i = 1; i < l; i++){
            if (ion == 1 || ion == 3){
                if(nums[i-1] < nums[i])
                continue;
                else if (nums[i-1] > nums[i])
                ion++;
                else
                return false;
            }
            else{
                if (nums[i-1] > nums[i])
                continue;
                else if (nums[i-1] < nums[i])
                ion++;
                else
                return false;
            }
        }
        return ion == 3;
    }
}

File: 3952-trionic-array-i/trionic-array-i.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        if len(nums) <= 3 or nums[1] < nums[0] or nums[len(nums)-1] < nums[len(nums)-2]:
            return False

        ion = 0
        for i in range(1, len(nums)):
            if ion == 0 or ion == 2:
                if nums[i-1] < nums[i]:
                    continue
                elif nums[i-1] == nums[i]:
                    return False
                else:
                    ion +=1
            elif ion == 1:
                if nums[i-1] > nums[i]:
                    continue
                elif nums[i-1] == nums[i]:
                    return False
                else:
                    ion +=1
        
        return ion == 2
            


        